### PR TITLE
avoid add missing member in declaration file

### LINKED ISSUE #29482: Do not  return 'declare' or 'add index signature' quick fixes for types that are part of lib.d.ts
**TypeScript Version:**  typescript@3.3.0-dev.20190118

**Search Terms:**
- quick fix
- code action
- lib.d.ts

**Code**

For the code:

```ts
async function a() {
    const response = await fetch(`https://example.com`);
   response.jsonX();
}
```

Trigger quickfixes on `jsonX`

**Bug**
`declare` and `add index` quick fixes returned. These modify the `dom.d.ts` declarations of fetch.

screen shot 2019-01-18 at 11 27 01 am

We should never try to edit the lib.d.ts files

**Playground Link:** 

**Related Issues:**

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #29482

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -1698,6 +1698,10 @@ namespace ts {
         return !!node && !!(node.flags & NodeFlags.JSDoc);
     }
 
+    export function isInDeclarationFile(node: Node | undefined): boolean {
```
[the remark this answers is not in the document]
Comment (weswigham) [reply src/compiler/utilities.ts]: Probably worth remembering that the original issue is specifically about the stdlib files (which generally shouldn't be editable), and not _all_ .d.ts files. Adding more props to declarations within your own project are fine - but adding props to a stdlib or node_modules sourced declaration is prooooobably wrong. Well, it's wrong unless the addition is done by making an augmentation in the current file 😛

[the remark this answers is not in the document]
Comment (Kingwl) [reply src/compiler/utilities.ts]: Probably only check file in node_modules?

[the remark this answers is not in the document]
Comment (RyanCavanaugh) [reply src/compiler/utilities.ts]: We should really apply a concrete notion of where a .d.ts file "came from" - lib, upstream composite project, node_modules/types, or a local "hand-authored" .d.ts file, only the last of which is a valid rename target

[the remark this answers is not in the document]
Comment (sheetalkamat) [reply src/compiler/utilities.ts]: `isSourceFileFromExternalLibrary` is that not good indication?

[the remark this answers is not in the document]
Comment (RyanCavanaugh) [reply src/compiler/utilities.ts]: Sounds like exactly what we want 😃

**Pull request merged**